sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -45,11 +45,11 @@ statement
     | ALTER DATABASE identifier SET DBPROPERTIES tablePropertyList     #setDatabaseProperties
     | DROP DATABASE (IF EXISTS)? identifier (RESTRICT | CASCADE)?      #dropDatabase
     | createTableHeader ('(' colTypeList ')')? tableProvider
-        (OPTIONS tablePropertyList)?
+        (OPTIONS optionParameterList)?
         (PARTITIONED BY partitionColumnNames=identifierList)?
         bucketSpec?                                                    #createTableUsing
     | createTableHeader tableProvider
-        (OPTIONS tablePropertyList)?
```

Comment:
@HyukjinKwon I am on holiday - so I am bit slow with my responses.

Yo have understood me correctly. What I am suggesting will affect the DBPROPERTIES and TBLPROPERTIES; it will also allow for boolean and numeric options. I don't think this is a bad thing, it is better to have a lenient parser and to constrain behavior in the `AstBuilder` (this allows us to throw much better error messages).